### PR TITLE
Refactor Inventory Model and Manager

### PR DESCRIPTION
**Pull request by proubatsis**
https://github.com/OpenLiberty/guides-common/issues/137

### CODE REVIEW

**Comment by proubatsis** [comment]
Since @andrewdes is using a hash map for his updated model, alternatively we could use a ConcurrentHashMap to solve this issue: https://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ConcurrentHashMap.html

cc @evelinec @Kubik42 @yeekangc

**Comment by Kubik42** [comment]
If ConcurrentHashMaps have built in synch, then I prefer using it instead

**Comment by proubatsis** [comment]
@evelinec @Kubik42 I updated this PR to incorporate @andrewdes changes fromhttps://github.com/OpenLiberty/guide-microprofile-fallback/pull/63 and changed `HashMap` to `ConcurrentHashMap`.

**Comment by proubatsis** [comment]
https://github.com/OpenLiberty/guide-microprofile-opentracing/pull/31

**Comment by evelinec** [comment]
@proubatsis had tested the same changes on these guides:
cdi-intro
microprofile-opentracing
microprofile-openapi (required a change in the openapi.yaml file as well)
microprofile-config
microprofile-health 
microprofile-jwt (required some changes on the frontend as well)
microprofile-metrics
microprofile-rest-client
microprofile-fallback

**Comment by yeekangc** [comment]
How does this affect MP OpenApi since we won't have the System object, @proubatsis @evelinec?

**Comment by evelinec** [comment]
@yeekangc take a look at the discussions in the openapi PR: https://github.com/OpenLiberty/guide-microprofile-openapi/pull/18

**Comment by yeekangc** [comment]
I already looked, @evelinec, and thank you.  There was no explanation on the why.  I would like to understand that and the external impact.  I can see the changes can work.

**Comment by proubatsis** [comment]
@yeekangc @evelinec @Kubik42 @andrewdes I changed it to use a `Map`, but the response type should remain the same now.

**Comment by proubatsis** [comment]
@Kubik42 @yeekangc @evelinec 
This summarizes my latest changes, is this more along the lines of what is being looked for?

* Extract `System` class into its own file and rename to `SystemModel` since it shares a name with the `System` class in the Java standard library
* Refactor `InventoryList` such that it is a simple representation of data rather than having any associated behaviour
* Refactor `InventoryManager` to take more responsibility for adding/removing systems and to synchronize access to the list. Also split `get()` into two methods: `get()` and `add()`.
* Adjust `InventoryResource` to use updated `InventoryManager` methods and to add some clarity

**Comment by proubatsis** [comment]
These are the changes that would be required in the text portions of the guides:
===

**cdi-intro**,	Handling Deps: get() method has now been split into add() and get()
**microprofile-opentracing**,	Enabling tracing: Change the addToInventoryList() span since it doesn't exist anymore
**microprofile-openapi**,	Generating Docs: Need reader to annotate InventoryList and SystemModel now since they're not in the same file
**microprofile-config**,	Adding Config: Augment add() and get() to accept port number and modify description accordingly
**microprofile-fallback**,	No changes
**microprofile-health**,	No changes
**microprofile-jwt**,	No changes
**microprofile-metrics**,	No changes
**microprofile-rest-client**, No changes
**hateoas**,	No changes

File: README.adoc [hunk 1]
```diff
@@ -102,8 +102,8 @@ Create the inventory manager class in the `src/main/java/io/openliberty/guides/i
 include::finish/src/main/java/io/openliberty/guides/inventory/InventoryManager.java[tags=**;!copyright;]
 ----
 
-This bean contains two simple functions. The `get()` function is for retrieving entries from calling the `system` service,
-and then add it to the inventory list.
+This bean contains three simple functions. The `get()` function is for retrieving entries from calling the `system` service.
+The `add()` function adds an entry to the inventory.
```
Comment (Kubik42) [inline README.adoc]: Can you keep the format the same? ie. write "The add() function is for adding entries to the inventory"

Comment (proubatsis) [reply README.adoc]: Ok, fixed.

**Comment by proubatsis** [comment]
`SystemModel` is renamed to `SystemData`. Is this all good now?

**Pull request merged**